user_tools/src/spark_rapids_tools/enums.py

```diff
@@ -53,7 +53,11 @@ def fromstring(cls, value: str) -> 'EnumeratedType':
         """
         attribute = getattr(cls, value.upper(), None)
         if attribute is None:
-            raise ValueError(f'{value} is not a valid {cls.__name__}')
+            # Call the enum constructor to call _missing_ in case no match
+            try:
+                return cls(value)
+            except ValueError as exc:
+                raise ValueError(f'{value} is not a valid {cls.__name__}') from exc
```

Comment:
I don't think we should do that here to solve the problem of mapping in some other configurations.